Adds a medkit wall cabinet and makes empty medkits printable.

shellspeed1 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
This pr adds a new medkit wall cabinet to the game that works the exact same way as a fire extinguisher cabinets . It also enables medbay to print empty medkits to put in them as well as to hold supplies for doctors. Both are unlocked with biological technology researched and are printable at the medical protolathe. If placed by mappers it will contain a basic medkit. Also thanks to BrokenOculus for making sprites. 
<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
This PR is meant to encourage medbay to distribute supplies around the station to make it easier to handle emergencies and give them something to do to help make medbay a little less reactive and a bit more proactive. Also helps to make supplies a bit more visible since people have a tendency to fail to look inside the chem fridge for medication the chemist has made.

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog
:cl:
add: Adds medkit wall cabinets and empty medkits to the medical protolathe.
add: Medkit wall cabinents and empty medkits are now part of biological technology.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

shellspeed1 committed:
Creates medkit wall cabinet structure.

shellspeed1 committed:
adds medkit cabinet and empty medkit to protolathe

shellspeed1 committed:
makes empty medkits and medkit wall cabinet researchable

shellspeed1 committed:
Sprites Thanks to BrokenOculus

shellspeed1 committed:
merge conflict

shellspeed1 committed:
removes medkit from seperate file

shellspeed1 committed:
Makes medkit subtype

shellspeed1 committed:
Updates research

shellspeed1 committed:
Merge branch 'master' into wallmounted-medkits-mk2

shellspeed1 committed:
OCD spacing in research

ExcessiveUseOfCobblestone commented:
all /obj's should be supported for going into a cabinet, it should just filter per type what is allowed to go  in

shellspeed1 commented:
> 
> 
> all /obj's should be supported for going into a cabinet, it should just filter per type what is allowed to go in

Not sure what you mean by that, would you mind explaining in a bit more detail?

Time-Green commented:
> > all /obj's should be supported for going into a cabinet, it should just filter per type what is allowed to go in
> 
> Not sure what you mean by that, would you mind explaining in a bit more detail?

He means "please add support for every item to be wall cabineted easily without having to completely rewrite everything"

ExcessiveUseOfCobblestone commented:
if I have to make a new var then rewrite the entire proc just to support particular objects (when just typing the var as /obj/ won't break anything) that's really yikes.

If I wanted a cabinet that added clownhorns all I should have to do is subtype then set the "what can go into this cabinet" var to clownhorns without having to do anything else.

shellspeed1 commented:
I'll see what I can do.

shellspeed1 committed:
Allowed item var

shellspeed1 committed:
Merge branch 'wallmounted-medkits-mk2' of https://github.com/shellspeed1/tgstation into wallmounted-medkits-mk2

shellspeed1 commented:
I managed to make allowed item a var, however having issues making a variable to set what item it spawns with.

shellspeed1 committed:
sprite issue

cacogen commented:
We used to have medical wall cabinets with shit in them that was moved to closets interestingly enough

shellspeed1 commented:
I was wondering why we didn't have them.

ExcessiveUseOfCobblestone commented on code/game/objects/structures/extinguisher.dm:
```diff
@@ -9,7 +9,11 @@
 	var/opened = FALSE
+	//set allowed item to determine what item is allowed inside.
+	var/alloweditem
+	alloweditem = /obj/item/extinguisher
```
bruhminium

ExcessiveUseOfCobblestone replied:
```DM
var/alloweditem = /obj/item/extinguisher
```

shellspeed1 replied:
aaaaaaaaaaa, fixed it.

ExcessiveUseOfCobblestone commented on code/game/objects/structures/extinguisher.dm:
```diff
@@ -9,7 +9,11 @@
 	integrity_failure = 50
 	var/obj/item/extinguisher/stored_extinguisher
```
```DM
var/obj/store_object
```

you'll need to rename all `stored_extinguisher` to `stored_object` tho

shellspeed1 replied:
why though?

shellspeed1 replied:
eh whatever, did it.

shellspeed1 committed:
The beatings will continue until morale improves.

Swaps stored extinguisher var for stored object as per request, also aaaaaaaaa missed an enter.

Poojawa commented:
I think this would benefit further by having a generic cabinet as a parent/basic frame. which would then turn into whichever applicable item is attached to it.

then swap all the paths from `/obj/structure/extinguisher_cabinet` to being `/obj/structure/item_cabinet`  

with `/obj/structure/item_cabinet/fire_extinguisher` , `/obj/structure/item_cabinet/medical` , `/obj/structure/item_cabinet/gun_rack`  etc etc

shellspeed1 commented:
> 
> 
> I think this would benefit further by having a generic cabinet as a parent/basic frame. which would then turn into whichever applicable item is attached to it.
> 
> then swap all the paths from `/obj/structure/extinguisher_cabinet` to being `/obj/structure/item_cabinet`
> 
> with `/obj/structure/item_cabinet/fire_extinguisher` , `/obj/structure/item_cabinet/medical` , `/obj/structure/item_cabinet/gun_rack` etc etc
> 
> you could then make defib racks attach in with this too.

I am unsure, partially because defibs wall mounts are an actual machine.

Poojawa commented:
Yee, Technically out of scope as well, but the thought occurred so

I believe they're only machinery so that they receive the effects of APC charging.

nemvar commented on code/game/objects/structures/extinguisher.dm:
```diff
@@ -142,13 +146,56 @@
+		icon_state = "medkit_empty"
+	else
+		//set this for item spawned inside
+		stored_object = new /obj/item/storage/firstaid/regular/(src)
```
Every medkit cabinet that gets spawned without building == TRUE has both a first aid kit AND an extinguisher in it. pls fix.

shellspeed1 replied:
I'm not sure how your getting that issue as on my end it isn't happening.

shellspeed1 replied:
If you can confirm that this is an issue, I'll take a look at it. After testing it again on my side still wasn't having issues.

nemvar commented on code/game/objects/structures/extinguisher.dm:
```diff
@@ -142,13 +146,56 @@
+		icon_state = "medkit_closed"
+		return
+	if(stored_object)
+		if(istype(stored_object, /obj/item/storage/firstaid/regular)) //leave this in case someone decides to add more sprites for different kits
```
If someone decides to add more sprites to this, they can just add this code themselves. I would remove it.

shellspeed1 replied:
made sprites vars so this no longer applies.

nemvar commented on code/game/objects/structures/extinguisher.dm:
```diff
@@ -142,13 +146,56 @@
+	if(building)
+		setDir(ndir)
+		pixel_x = (dir & 3)? 0 : (dir == 4 ? -27 : 27)
+		pixel_y = (dir & 3)? (dir ==1 ? -30 : 30) : 0
```
Make the pixel_x/pixel_y thing variables on the item and get rid of this entire duplicate code block.

shellspeed1 replied:
No longer applies. Anyway that just sets which side of the wall its on if I understand.

shellspeed1 committed:
turned more things into vars.

ExcessiveUseOfCobblestone commented on code/game/objects/structures/extinguisher.dm:
```diff
@@ -7,41 +7,53 @@
 	max_integrity = 200
 	integrity_failure = 50
-	var/obj/item/extinguisher/stored_extinguisher
+	var/obj/item/extinguisher/stored_object
```
```suggestion
	var/obj/item/stored_object
```

shellspeed1 committed:
Update code/game/objects/structures/extinguisher.dm

Co-Authored-By: moo <11748095+ExcessiveUseOfCobblestone@users.noreply.github.com>

shellspeed1 commented:
Fuck, missed that one.

shellspeed1 commented:
Meh, I might redo this later. Seems like people would like to limit the availability of medical supplies for right now anyway.
